src/gpu/nvidia/cudnn_matmul_lt_impl.hpp

```diff
@@ -0,0 +1,726 @@
+                    weight_size_, 1, 256);
+        }
+        if (source_size_ > 0) {
+            scratchpad.book(memory_tracking::names::key_matmul_lt_block_b,
+                    source_size_, 1, 256);
+        }
+        if (dest_size_ > 0) {
+            scratchpad.book(memory_tracking::names::key_matmul_lt_block_c,
+                    dest_size_, 1, 256);
+        }
+    }
+
+    void execute(cublasHandle_t cublas_handle, cudnnHandle_t cudnn_handle,
+            void *a, void *b, void *c, void *bias, void *algo_scratch,
+            void *reorder_scratch, void *block_a_scratch, void *block_b_scratch,
+            void *block_c_scratch, void *src_scale, void *wei_scale,
+            void *dst_scale) override {
+
+        cudaStream_t streamId;
+        auto lt_handle = (cublasLtHandle_t)(cublas_handle);
+        CUBLAS_EXECUTE_FUNC(cublasGetStream, cublas_handle, &streamId);
+
+        if (imma_ampere_case_) {
+            transform_matrix(lt_handle, b_layout_, b, blocked_b_layout_,
+                    block_b_scratch, !trans_b_, streamId);
+            b = block_b_scratch;
+            if (!w_blocked_) {
+                transform_matrix(lt_handle, a_layout_, a, blocked_a_layout_,
+                        block_a_scratch, trans_a_, streamId);
+                a = block_a_scratch;
+            }
+        }
+
+        cublasLtEpilogue_t epilogue = CUBLASLT_EPILOGUE_DEFAULT;
+        if (with_bias_) {
+            if (with_relu_) {
+                epilogue = CUBLASLT_EPILOGUE_RELU_BIAS;
+            } else {
+                epilogue = CUBLASLT_EPILOGUE_BIAS;
+            }
+            CUBLAS_EXECUTE_FUNC(cublasLtMatmulDescSetAttribute, operation_desc_,
+                    CUBLASLT_MATMUL_DESC_BIAS_POINTER, &bias, sizeof(bias));
+        } else if (with_relu_ && !with_bias_) {
+            epilogue = CUBLASLT_EPILOGUE_RELU;
+        }
+        CUBLAS_EXECUTE_FUNC(cublasLtMatmulDescSetAttribute, operation_desc_,
+                CUBLASLT_MATMUL_DESC_EPILOGUE, &epilogue, sizeof(epilogue));
+
+        float scale = 1.0f;
+        float host_dst_scale = 1.0f;
+        if (src_scale) {
+            float host_src_scale = 1.0f;
+            CUDA_EXECUTE_FUNC(cuMemcpyAsync, (CUdeviceptr)&host_src_scale,
+                    (CUdeviceptr)src_scale, sizeof(float), streamId);
+            scale *= host_src_scale;
+        }
+        if (wei_scale) {
+            float host_wei_scale = 1.0f;
+            CUDA_EXECUTE_FUNC(cuMemcpyAsync, (CUdeviceptr)&host_wei_scale,
+                    (CUdeviceptr)wei_scale, sizeof(float), streamId);
+            scale *= host_wei_scale;
+        }
+        if (dst_scale) {
+            CUDA_EXECUTE_FUNC(cuMemcpyAsync, (CUdeviceptr)&host_dst_scale,
+                    (CUdeviceptr)dst_scale, sizeof(float), streamId);
+            // For eltwise post-ops, apply the dst scale afterward
+            if (!with_separate_eltwise_) scale /= host_dst_scale;
+        }
```

Comment:
I believe cublasLT can be configured to take device pointers (CUBLASLT_POINTER_MODE_DEVICE)

Reply:
That is true, however `src`,`wei` and `dst` scale map to the cublasLT matmul `alpha` param and the post op sum maps to `beta` i do not believe using the device pointer mode would work in this case apart from creating a sycl kernel to do
`alpha = (1 * src_scale * wei_scale) / dst_scale` on device side before passing to the matmul.
Would that be the preferred approach?

Reply:
It would work with oneDNN semantics if only this implementation won't support post-ops. If it would, scales should be separated as src-wei scales are applied before post-ops and dst scales are applied at the end.

Reply:
Latest commits added src/weight scale for multiple values as well as dst scaling applied after post-ops if any